cmd/client/wallet/main.go

```diff
@@ -767,11 +767,12 @@ func processTransferCommand() {
 
 	fmt.Printf("Unlock account succeeded! '%v'\n", senderPass)
 
-	chainConfig := params.MainnetChainConfig
-	if walletProfile.Profile != defaultProfile {
-		chainConfig = params.TestnetChainConfig
+	chainID, ok := new(big.Int).SetString(walletProfile.ChainID, 0)
+	if !ok {
+		fmt.Printf("invalid chain ID %#v in config", walletProfile.ChainID)
+		return
 	}
-	tx, err = ks.SignTx(account, tx, chainConfig.ChainID)
+	tx, err = ks.SignTx(account, tx, chainID)
```

Comment:
Where is the change to wallet profile file to add the chainid ?

Reply:
If a chain ID is absent, the profile reader infers the chain ID from the profile name: 1 for `main` and `default`, 3 for `pangaea`, and 2 for everything else.